content/best_practices.adoc

```diff
@@ -56,7 +56,48 @@ Some operations require logging/auditing to allow to understand what happened du
+
+Knowing that, the Vue applications rendering has to be made in an asynchrnous way without impacting other applications present in the page. For example, when having information to retrieve from Server-End using REST API calls, a fetch call will be used. To ensure to have a good UX, a loading effect in the application itself has to be rendered first time the Vue application is mounted, until all needed information are retrieved.
+
+But, using the fetch API to retrieve information asynchronously from Server-End has a big drawback. In fact, a Browser allows a limited concurrent HTTP requests for each domain (https://blog.stackpath.com/glossary-domain-sharding/[Domain Sharding]). In a web page, that holds a lot of Vue applications, each one will retrieve information from Server-End using different REST endpoints. This can consequently lead to a very bad UX with a considerable idle time for HTTP requests. Fortunately, the Web Standards has evolved to take into consideration this complexity of new Web Applications especially by introducing the `HTTP/2` protocol (https://www.limelightonline.co.nz/blog/what-http2-benefit-your-website/[Read more about benifits]). To benefit from it, and to allow to increase the parallel concurrent requests to send to Server-End, we will have to `preload` and `prefetch` mandatory information that renders the Web Page (https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf[Read more]).
+
+The retrieved HTML DOM of eXo Platform contains references to the needed CSS files (computed from the list of Portlets being part of the displayed page). The fact that we have CSS links, in the retrieved DOM, will help browser to determine which files to retrieve using `HTTP/2` in an early stage of Web Page rendering phase. But the Javascript files uses a different mechanism based on https://requirejs.org/[RequireJS], that will retrieve JS files at the end of Web Page processing. This will lead to send multiple HTTP requests that will not be optimized using `HTTP/2`. Thus, the JS and CSS files are preloaded to ensure retrieving those files once at the same time, even when having tens of them. Consequently, all JS files has to use https://docs.exoplatform.org/en/6.1/JavaScript.html#amd-and-requirejs[AMD] mechanism to be imported inside a page and any *new module has to be imported as a dependency of a Portlet asset* to ensure not losing time in Web Page rendering phase. So, there are several *bad practices* to avoid such as, adding a `link` element inside a Vue application, or even worse when using `requirejs` in render phase on a module that is not a dependency of a displayed Portlet.
```

Comment:
```suggestion

The retrieved HTML DOM of eXo Platform contains references to the needed CSS files (computed from the list of Portlets being part of the displayed page). The fact that we have CSS links, in the retrieved DOM, will help the browser to determine which files to retrieve using `HTTP/2` in an early stage of the Web Page rendering phase. But the Javascript files use a different mechanism based on https://requirejs.org/[RequireJS], which will retrieve JS files at the end of Web Page processing. This will lead to sending multiple HTTP requests that will not be optimized using `HTTP/2`. Thus, the JS and CSS files are preloaded to ensure retrieving those files once at the same time, even when having tens of them. Consequently, all JS files have to use https://docs.exoplatform.org/en/6.1/JavaScript.html#amd-and-requirejs[AMD] mechanism to be imported inside a page and any *new module has to be imported as a dependency of a Portlet asset* to ensure not losing time in the Web Page rendering phase. So, there are several *bad practices* to avoid such as, adding a `link` element inside a Vue application, or even worse when using `requirejs` in the render phase on a module that is not a dependency of a displayed Portlet.
```